### PR TITLE
docs: fix PHPDocs for `View`

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
See #8732

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/View/View.php [hunk 1]
```diff
@@ -487,6 +490,8 @@ public function include(string $view, ?array $options = null, $saveData = true):
     /**
      * Returns the performance data that might have been collected
      * during the execution. Used primarily in the Debug Toolbar.
+     *
+     * @return list<float|string>
```
Comment (kenjis) [inline system/View/View.php:494]:
```suggestion
     * @return list<array{start: float, end: float, view: string}>
```

File: system/View/View.php [hunk 2]
```diff
@@ -86,7 +86,7 @@ class View implements RendererInterface
      * Cache stats about our performance here,
      * when CI_DEBUG = true
      *
-     * @var array
+     * @var list<float|string>
```
Comment (kenjis) [inline system/View/View.php:89]:
```suggestion
     * @var list<array{start: float, end: float, view: string}>
```

File: system/View/Table.php [hunk 3]
```diff
@@ -27,21 +27,21 @@ class Table
...
-     * @var list<array>|list<list<array>>
+     * @var list<array<string, string>>|list<list<array<string, string>>>
      */
     public $rows = [];
 
     /**
      * Data for table heading
      *
-     * @var array
+     * @var array<int, mixed>
```
Comment (kenjis) [inline system/View/Table.php:37]: Is it possible that the element value is not a string?

File: system/View/Table.php [hunk 4]
```diff
@@ -27,21 +27,21 @@ class Table
...
      */
     public $heading = [];
 
     /**
      * Data for table footing
      *
-     * @var array
+     * @var array<int, mixed>
```
Comment (kenjis) [inline system/View/Table.php:44]: Is it possible that the element value is not a string?

File: system/View/Table.php [hunk 5]
```diff
@@ -297,7 +300,7 @@ public function setCaption($caption)
     /**
      * Generate the table
      *
-     * @param array|BaseResult|null $tableData
+     * @param array<int, mixed>|BaseResult|null $tableData
```
Comment (kenjis) [inline system/View/Table.php:303]: Like this?
```suggestion
     * @param BaseResult|list<array<int|string, float|int|string>>|null $tableData
```

`var_export($tableData, true)`:
```
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    0 => 'Name',
    1 => 'Color',
    2 => 'Size',
  ),
  1 => 
  array (
    0 => 'Fred',
    1 => 'Blue',
    2 => 'Small',
  ),
  2 => 
  array (
    0 => 
    array (
      'data' => 'Mary',
      'class' => 'cssMary',
    ),
    1 => 
    array (
      'data' => 'Red',
      'class' => 'cssRed',
    ),
    2 => 
    array (
      'data' => 'Large',
      'class' => 'cssLarge',
    ),
  ),
  3 => 
  array (
    0 => 'John',
    1 => 'Green',
    2 => 'Medium',
  ),
)
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    0 => 'Name',
    1 => 'Color',
    2 => 'Size',
  ),
  1 => 
  array (
    0 => 'Fred',
    1 => 'Blue',
    2 => 'Small',
  ),
  2 => 
  array (
    0 => 'Mary',
    1 => 'Red',
    2 => NULL,
  ),
  3 => 
  array (
    0 => 'John',
    1 => 'Green',
    2 => 'Medium',
  ),
)
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    0 => 'Name',
    1 => 'Color',
    2 => 'Size',
  ),
  1 => 
  array (
    0 => 'Fred',
    1 => 'Blue',
    2 => 'Small',
  ),
  2 => 
  array (
    0 => 'Mary',
    1 => 'Red',
    2 => 'Large',
  ),
  3 => 
  array (
    0 => 'John',
    1 => 'Green',
    2 => 'Medium',
  ),
)
DEBUG - 2024-05-06 01:06:12 --> CodeIgniter\View\DBResultDummy::__set_state(array(
   'connID' => NULL,
   'resultID' => NULL,
   'resultArray' => 
  array (
  ),
   'resultObject' => 
  array (
  ),
   'customResultObject' => 
  array (
  ),
   'currentRow' => 0,
   'numRows' => NULL,
   'rowData' => NULL,
))
DEBUG - 2024-05-06 01:06:12 --> array (
)
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> NULL
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    'name' => 'Max',
    'age' => 30,
    'id' => 5,
  ),
)
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    'name' => 'Fred',
    'age' => 30,
    'id' => 5,
  ),
)
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    'name' => 'Fred',
    'age' => 30,
    'id' => 5,
  ),
)
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    'name' => 'Fred',
    'id' => 5,
  ),
)
DEBUG - 2024-05-06 01:06:12 --> array (
  0 => 
  array (
    'id' => 1,
    'id_cliente' => 1,
    'codigo' => 'codigo1',
    'data' => '2023-10-16 21:53:25',
    'tipo_desconto' => 'NENHUM',
    'valor_desconto' => '',
    'created_at' => '2023-10-16 21:53:25',
    'updated_at' => '2023-10-16 21:53:25',
    'deleted_at' => '',
  ),
  1 => 
  array (
    'id' => 2,
    'id_cliente' => 2,
    'codigo' => 'codigo2',
    'data' => '2023-10-16 21:53:25',
    'tipo_desconto' => 'REAL',
    'valor_desconto' => 10.0,
    'created_at' => '2023-10-16 21:53:25',
    'updated_at' => '2023-10-16 21:53:25',
    'deleted_at' => '',
  ),
  2 => 
  array (
    'id' => 3,
    'id_cliente' => 3,
    'codigo' => 'codigo3',
    'data' => '2023-10-16 21:53:25',
    'tipo_desconto' => 'PERCENTUAL',
    'valor_desconto' => 10.0,
    'created_at' => '2023-10-16 21:53:25',
    'updated_at' => '2023-10-16 21:53:25',
    'deleted_at' => '',
  ),
)
```

File: system/View/Table.php [hunk 6]
```diff
@@ -157,10 +158,10 @@ public function setFooting()
      * columns. This allows a single array with many elements to be
      * displayed in a table that has a fixed column count.
      *
-     * @param array $array
-     * @param int   $columnLimit
+     * @param list<string> $array
+     * @param int          $columnLimit
      *
-     * @return array|false
+     * @return array<int, mixed>|false
```
Comment (kenjis) [inline system/View/Table.php:164]: Like this?
```suggestion
     * @return list<string>|list<list<string>>|false
```

File: system/View/Table.php [hunk 7]
```diff
@@ -472,7 +475,7 @@ protected function _setFromDBResult($object)
     /**
      * Set table data from an array
      *
-     * @param array $data
+     * @param array<int, mixed> $data
```
[the remark this answers is not in the document]
Comment (ddevsr) [reply system/View/Table.php:478]: I see in debug, have 2 nested

```diff
- @param list<array<int|string, float|int|string>> $data
+ @param list<array<int|string, float|int|string|list<string>>> $data
```

[the remark this answers is not in the document]
Comment (kenjis) [reply system/View/Table.php:478]: Oh, this is `list<array<int, string|array<string, string>>>`.
```
array (
  0 => 
  array (
    0 => 'Name',
    1 => 'Color',
    2 => 'Size',
  ),
  1 => 
  array (
    0 => 'Fred',
    1 => 'Blue',
    2 => 'Small',
  ),
  2 => 
  array (
    0 => 
    array (
      'data' => 'Mary',
      'class' => 'cssMary',
    ),
    1 => 
    array (
      'data' => 'Red',
      'class' => 'cssRed',
    ),
    2 => 
    array (
      'data' => 'Large',
      'class' => 'cssLarge',
    ),
  ),
  3 => 
  array (
    0 => 'John',
    1 => 'Green',
    2 => 'Medium',
  ),
)
```

[the remark this answers is not in the document]
Comment (kenjis) [reply system/View/Table.php:478]: Can this be nested indefinitely?
https://github.com/codeigniter4/CodeIgniter4/blob/299bb8b9e6e09d308cdf93c5d7236702b17e313e/tests/system/View/TableTest.php#L399

**Review by kenjis — APPROVED** [review]
Frankly, I am not sure of the exact type of the Table class. 
However, I think this PR is not wrong, so I will merge it.

**Pull request merged**